chat/user.go

```diff
@@ -36,6 +36,19 @@ func NewUser(name string) *User {
 	return &u
 }
 
+// NewUserWithConfig creates a new User with the specified UserConfig
+func NewUserWithConfig(name string, config UserConfig) *User {
```

Comment:
Why not just shove the Config onto the user after the fact? It's a public attribute and all. Otherwise it's an identical constructor. :)